influxdb_iox/tests/end_to_end_cases/debug.rs

```diff
@@ -134,6 +117,30 @@ async fn build_catalog() {
     .await
 }
 
+/// Rebuilds a catalog from an export directory, starts up a server
+/// and verifies the running `sql` in `namespace` produces `expected`
+async fn rebuild_and_query(table_dir: &Path, namespace: &str, sql: &str, expected: &[&str]) {
+    // Very occassionally, something goes wrong with the sqlite based
+    // catalog and it doesn't get the new files. Thus try a few times
+    //
+    // See https://github.com/influxdata/influxdb_iox/issues/8203
```

Comment:
This is basically the same approach as we tried before (retries) except that it retries the entire test (rather than just the query) as there is something wrong with the catalog in rare occasions

Reply:
"something wrong" doesn't sound super convincing :sweat_smile:

Reply:
I filed https://github.com/influxdata/influxdb_iox/issues/8287 to track it (not any more convincing, but perhaps more explicit)

```suggestion
    // See https://github.com/influxdata/influxdb_iox/issues/8287
```